packages/ag-charts-enterprise/src/error-bar/errorBar.ts

```diff
@@ -0,0 +1,138 @@
+    yUpperName?: string = undefined;
+
+    private readonly cartesianSeries: _ModuleSupport.CartesianSeries<any, any>;
+    private readonly groupNode: _Scene.Group;
+    private readonly selection: _Scene.Selection<ErrorBarNode>;
+    private nodeData: (ErrorBarPoints | undefined)[] = [];
+
+    constructor(ctx: _ModuleSupport.SeriesContext) {
+        super();
+
+        const supportedSeriesTypes = ['line'];
+        if (!supportedSeriesTypes.includes(ctx.series.type)) {
+            throw new Error(
+                `AG Charts - unsupported series type '${
+                    ctx.series.type
+                }', error bars supported series types: ${supportedSeriesTypes.join(', ')}`
+            );
+        }
+        this.cartesianSeries = ctx.series as CartesianSeries<any, any>;
+        const { contentGroup } = this.cartesianSeries;
+
+        this.groupNode = new _Scene.Group({ name: `${contentGroup.id}-series-errorBars` });
+        contentGroup.appendChild(this.groupNode);
+        this.selection = _Scene.Selection.select(this.groupNode, () => this.errorBarFactory());
+
+        const listener = this.cartesianSeries.addListener('data-model', (event) => this.onDataProcessed(event));
+        this.destroyFns.push(() => this.cartesianSeries.removeListener(listener));
+    }
+
+    onDataProcessed(event: {
+        dataModel: _ModuleSupport.DataModel<any, any, any>;
+        processedData: _ModuleSupport.ProcessedData<any>;
+    }) {
+        if (event.dataModel !== undefined && event.processedData !== undefined) {
+            this.createNodeData(event.dataModel, event.processedData);
+            this.update();
+        }
+    }
+
+    createNodeData(
+        dataModel: _ModuleSupport.DataModel<any, any, any>,
+        processedData: _ModuleSupport.ProcessedData<any>
+    ) {
+        const { yLowerKey, yUpperKey, nodeData } = this;
+        const xIndex = dataModel?.resolveProcessedDataIndexById(this.cartesianSeries, `xValue`).index;
+        const xScale = this.cartesianSeries.axes[ChartAxisDirection.X]?.scale;
+        const yScale = this.cartesianSeries.axes[ChartAxisDirection.Y]?.scale;
+        if (!xScale || !yScale || !yLowerKey || !yUpperKey || xIndex === undefined) {
+            return;
+        }
+
+        const convert = (scale: _Scale.Scale<any, any, any>, value: any) => {
+            const offset = (scale.bandwidth ?? 0) / 2;
+            return scale.convert(value) + offset;
+        };
+
+        nodeData.length = processedData.data.length;
+
+        for (let i = 0; i < processedData.data.length; i++) {
+            const { datum, values } = processedData.data[i];
+            const xDatum = values[xIndex];
+
+            if (yLowerKey in datum && yUpperKey in datum) {
+                nodeData[i] = {
+                    yLowerPoint: { x: convert(xScale, xDatum), y: convert(yScale, datum[yLowerKey]) },
+                    yUpperPoint: { x: convert(xScale, xDatum), y: convert(yScale, datum[yUpperKey]) },
```

Comment:
Although accessing the `datum` properties directly here achieves the right outcome for line, I think it would be better to specify additional properties to the `DataModel` so that grouped/stacked cases can be handled correctly.

Maybe we can follow-up on this in the next PR?

Reply:
It's unclear to me what the group/stacked cases involve. Let's follow up on this during a 1-on-1.